### PR TITLE
modernize `tbg`

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- rename some variables, no need to start variables with `tooltpl` anymore since we removed a lot of our ugly `sed` magic in #4905
- use `$(command)` indtead of `'comamnd'`

### CODE REVIEW

**Review by PrometheusPi — DISMISSED** [review]
Looks good to me - will test asap

**Comment by psychocoderHPC** [comment]
This will give a small overview about both ways to execute an command https://www.redhat.com/sysadmin/backtick-operator-vs-parens

**Comment by PrometheusPi** [comment]
If I run my test from this morning, I get a `grep` warning:
```
grep: character class syntax is [[:space:]], not [:space:]
```
and it takes longer, now 28s.

File: bin/tbg [hunk 1]
```diff
@@ -208,15 +207,15 @@ function check_final
...
+    not_replaced=$(grep -o "\![[:alpha:]][[:alnum:]_]*" $final_file | sort | uniq)
+    not_replaced_cnt=$(echo $not_replaced | wc -w)
 
     if [ $not_replaced_cnt -gt 0 ] ; then
         echo "ERROR: $not_replaced_cnt variable(s) _not_ replaced from template (tpl):"
         echo $not_replaced
 
         #create an OR concated pattern
-        nrv_or=`echo $not_replaced | sed 's/[[:space:]]/|/g'`
+        nrv_or=$(echo $not_replaced | sed 's/[[:space:]]/|/g')
```
Comment (PrometheusPi) [inline bin/tbg:218]: That's most likely the line causing the warning

Comment (psychocoderHPC) [reply bin/tbg:218]: I fixed it. It was coming from `egrep` not sed. I do not fully understand why `[[:blank:]]` is throwing the warning.
I switched to the old-school way of defining a list of characters.

Comment (psychocoderHPC) [reply bin/tbg:218]: https://github.com/ComputationalRadiationPhysics/picongpu/pull/4906/files#diff-7b05b982bf71e63ce4dd3f2fd29332ae3e3023f0357d503da6d0f011d936a155R393 is the problematic line

**Comment by chillenzer** [comment]
I'm not familiar enough with the code of this to see if there are any bad things hidden under a thick layer of trivial changes. Two suggestions:
1. Splitting this into several commits might help reviewing this.
2. Most of the changes I've seen are not something anyone should be doing by hand. Should we adapt [shellcheck](https://github.com/koalaman/shellcheck). Not sure if this one also automatically applies fixes but probably something like that exists.

File: bin/tbg (hunk 1 above)
Comment (psychocoderHPC) [reply bin/tbg:218]: @PrometheusPi this is already solved in the PR

**Comment by psychocoderHPC** [comment]
> 1. Splitting this into several commits might help reviewing this.

All changes are in one file splitting it now will only introduce errors.

> 2. Most of the changes I've seen are not something anyone should be doing by hand. Should we adapt [shellcheck](https://github.com/koalaman/shellcheck). Not sure if this one also automatically applies fixes but probably something like that exists.

Yes, we can do it independent of this PR.

File: bin/tbg (hunk 1 above)
Comment (PrometheusPi) [reply bin/tbg:218]: I checked that I used your (new?) version with `nrv_or=$(echo $not_replaced | sed 's/[[:space:]]/|/g')` but I still get the warning:
```
grep: character class syntax is [[:space:]], not [:space:]
```

**Review by PrometheusPi — CHANGES_REQUESTED** [review]

File: bin/tbg (hunk 1 above)
Comment (psychocoderHPC) [reply bin/tbg:218]: offline checked: @PrometheusPi did a mistake and used the old version and not the latest PR.

Comment (PrometheusPi) [reply bin/tbg:218]: Yes, I pulled something from GitHub, but not the newest version. Tested it now, and it does not show the warning anymore.

**Review by PrometheusPi — APPROVED** [review]

**Pull request merged**